### PR TITLE
feat: color function (#2054)

### PR DESCRIPTION
**Pull request by Denis-Shamakin**
Added the ability to specify a function instead of color tables

### CODE REVIEW

File: typescript/packages/well-log-viewer/src/components/WellLogView.tsx [hunk 1]
```diff
@@ -827,6 +827,20 @@ function fillPlotTemplate(
     const scale =
         templateTrack?.scale !== options.scale ? options.scale : undefined;
 
+    let colorTableName = "";
+    if (typeof options.colorTable === "function") {
+        colorTableName = "Function";
+    } else if (options.colorTable) {
```
Comment (w1nklr) [inline typescript/packages/well-log-viewer/src/components/WellLogView.tsx:833]: replace by:
} else {
colorTableName = options.colorTable?.name ?? "";
}

File: typescript/packages/well-log-viewer/src/components/WellLogView.tsx [hunk 2]
```diff
@@ -827,6 +827,20 @@ function fillPlotTemplate(
...
+        colorTableName = options.colorTable.name;
+    }
+
+    let inverseColorTableName = "";
+    if (typeof options.inverseColorTable === "function") {
+        inverseColorTableName = "Function";
+    } else if (options.inverseColorTable) {
```
Comment (w1nklr) [inline typescript/packages/well-log-viewer/src/components/WellLogView.tsx:840]: replace by:
} else {
colorTableName = options.inverseColorTable?.name ?? "";
}

**Review by w1nklr — CHANGES_REQUESTED** [review]

File: typescript/packages/well-log-viewer/src/components/WellLogView.tsx (hunk 1 above)
Comment (Denis-Shamakin) [reply typescript/packages/well-log-viewer/src/components/WellLogView.tsx]: changed it to a conditional (ternary) operator:

```
        colorTable:
            typeof options.colorTable === "function"
                ? "Function"
                : options.colorTable?.name ?? "",
```

File: typescript/packages/well-log-viewer/src/components/WellLogView.tsx (hunk 2 above)
Comment (Denis-Shamakin) [reply typescript/packages/well-log-viewer/src/components/WellLogView.tsx]: changed it to a conditional (ternary) operator:

```
        inverseColorTable:
            typeof options.inverseColorTable === "function"
                ? "Function"
                : options.inverseColorTable?.name ?? "",
```

**Review by w1nklr — APPROVED** [review]

**Pull request merged**